service_info_cms/views.py

```
@@ -1,24 +1,79 @@
+import json
+import logging
+
 from cms.models.pagemodel import Page
+from django.conf import settings
+from django.http import HttpResponseBadRequest, HttpResponseServerError
 from django.shortcuts import redirect
+import requests
+
 from .models import PageRating
 
+GOOGLE_CAPTCHA_VERIFY_URL = 'https://www.google.com/recaptcha/api/siteverify'
+
+logger = logging.getLogger('__name__')
+
+
+def verify_captcha(secret, response, remoteip=None):
+    """
+    From https://developers.google.com/recaptcha/docs/verify:
+        secret: The shared key between your site and ReCAPTCHA.
+        response: The user response token provided by the reCAPTCHA to the user
+            and provided to your site on.
+        remoteip (optional): The user's IP address.
+    """
+    post_data = {
+        'secret': secret,
+        'response': response,
+    }
+    if remoteip:
+        post_data['remote'] = remoteip
+
+    try:
+        r = requests.post(
+            GOOGLE_CAPTCHA_VERIFY_URL,
+            data=post_data
+        )
+        if r.status_code != 200:
+            logger.error('Google reCAPTCHA -> %d', r.status_code)
+            return False
+        # r.content (bytes) vs. r.text (str):
+        #   JSON parsing requires str,
+        #   logging of bytes is better because it will all be on one line
+        #     (e.g., as b'{\n  "success": true\n}')
+        logger.debug('Google reCAPTCHA -> %s', r.content)
+        result = json.loads(r.text)
+        if not result['success']:
+            logger.error('Google reCAPTCHA -> %s', r.content)
+            return False
+    except:
```

Maybe make this bare except clause into `except Exception:` ?